pkg/kubelet/images/image_gc_manager.go

```diff
@@ -248,8 +252,15 @@ func (im *realImageGCManager) detectImages(ctx context.Context, detectTime time.
 		// New image, set it as detected now.
 		if _, ok := im.imageRecords[image.ID]; !ok {
 			klog.V(5).InfoS("Image ID is new", "imageID", image.ID)
-			im.imageRecords[image.ID] = &imageRecord{
-				firstDetected: detectTime,
+			if !utilfeature.DefaultFeatureGate.Enabled(features.RuntimeClassInImageCriAPI) {
+				im.imageRecords[image.ID] = &imageRecord{
+					firstDetected: detectTime,
```

Comment:
note: this works when the differentiator for the container runtime - runtime class config is a platform/version change that results in a different digest value (sha256) because at least one bit in the binary for that platform/version is different, but it does not work, I don't believe, when the differentiator for the runtime class is the snapshotter. 

As discussed in the KEP, one would need to create a tuple (imageName/imageDigest + runtimeClass) here to allow for identification if the image is ready based on this entry in the list.

Reply:
resolution may be as simple as return the differentiating fields in the image pull response/list apis.. and concatenate them in the ID as a parseable string... thus allowing the ID+runtime handler platform/version+snapshotter/fs to be used as the index..

Reply:
@mikebrow I am not sure I understand the question/concern here. Could you please clarify?
As mentioned in KEP 4216, we are not aiming to support snapshotter per runtime class at all with this kep. Here were are just going to be identifying image as name/ID and runtime class

Reply:
per discussion we agreed a TODO may be fine for extending to the "edited" runtime handler config situation, as you said, we can just require a restart to refresh ... and thus be able to trust two runtime handler configs will have two different names or "" and a non default name..

Reply:
@mikebrow  summarized the discussion we had here - https://github.com/kubernetes/kubernetes/pull/121456#discussion_r1372104416

Reply:
Addressed this in the new changes that I just pushed. Please let me know if it looks ok. Thanks!